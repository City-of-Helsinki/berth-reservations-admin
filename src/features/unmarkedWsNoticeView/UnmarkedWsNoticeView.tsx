import React from 'react';
import { useTranslation } from 'react-i18next';
import { PureQueryOptions } from 'apollo-client';

import styles from './unmarkedWsNoticeView.module.scss';
import Card from '../../common/card/Card';
import CardBody from '../../common/cardBody/CardBody';
import CardHeader from '../../common/cardHeader/CardHeader';
import CustomerProfileCard, { CustomerProfileCardProps } from '../../common/customerProfileCard/CustomerProfileCard';
import PageTitle from '../../common/pageTitle/PageTitle';
import PageContent from '../../common/pageContent/PageContent';
import LinkApplicationToCustomerContainer, {
  LinkApplicationToCustomerContainerProps,
} from '../linkApplicationToCustomer/LinkApplicationToCustomerContainer';
import Button from '../../common/button/Button';
import UnmarkedWsNoticeDetails, {
  UnmarkedWsNoticeDetailsProps,
} from '../unmarkedWsNoticeDetails/UnmarkedWsNoticeDetails';
import ActionHistoryCard from '../../common/actionHistoryCard/ActionHistoryCard';
import Grid from '../../common/grid/Grid';
import InvoiceCard from '../invoiceCard/InvoiceCardContainer';
import { Order } from '../invoiceCard/types';
import DeleteButton from '../../common/deleteButton/DeleteButton';
import { LeaseStatus } from '../../@types/__generated__/globalTypes';
import { canDeleteLease } from '../../common/utils/leaseUtils';
import ApplicationHeader from '../../common/applicationHeader/ApplicationHeader';

export interface UnmarkedWsNoticeViewProps {
  customerProfile: CustomerProfileCardProps | null;
  noticeDetails: UnmarkedWsNoticeDetailsProps;
  leaseStatus: LeaseStatus | null;
  order: Order | null;
  refetchQueries: PureQueryOptions[] | string[];
  winterStorageNotice: LinkApplicationToCustomerContainerProps['application'];
  isDeleteNoticeLoading: boolean;
  isCreateLeaseLoading: boolean;
  isDeleteLeaseLoading: boolean;
  handleCreateLease(): void;
  handleDeleteLease(): void;
  handleDeleteNotice(): void;
  handleEditCustomer(): void;
  handleLinkCustomer(customerId: string): void;
  handleUnlinkCustomer(): void;
}

const UnmarkedWsNoticeView = ({
  customerProfile,
  noticeDetails,
  order,
  leaseStatus,
  refetchQueries,
  winterStorageNotice,
  handleCreateLease,
  handleDeleteNotice,
  handleDeleteLease,
  handleEditCustomer,
  handleLinkCustomer,
  isDeleteNoticeLoading,
  isCreateLeaseLoading,
  isDeleteLeaseLoading,
  handleUnlinkCustomer,
}: UnmarkedWsNoticeViewProps) => {
  const { t } = useTranslation();

  return (
    <PageContent className={styles.noticeView}>
      <div className={styles.fullWidth}>
        <PageTitle title={t('unmarkedWsNotices.view.title')} />
      </div>

      <ApplicationHeader
        text={t('applicationList.applicationType.notice')}
        createdAt={noticeDetails.createdAt}
        status={noticeDetails.status}
        customerId={noticeDetails.customerId}
        handleUnlinkCustomer={handleUnlinkCustomer}
        handleDeleteApplication={handleDeleteNotice}
        isDeletingApplication={isDeleteNoticeLoading}
      />

      {customerProfile ? (
        <>
          <CustomerProfileCard {...customerProfile} handleEditCustomer={handleEditCustomer} />
          <ActionHistoryCard />
        </>
      ) : (
        <LinkApplicationToCustomerContainer application={winterStorageNotice} handleLinkCustomer={handleLinkCustomer} />
      )}

      <Card className={styles.fullWidth}>
        <CardHeader title={t('unmarkedWsNotices.noticeDetails.title').toUpperCase()} />
        <CardBody>
          <Grid colsCount={3}>
            <UnmarkedWsNoticeDetails {...noticeDetails} />
            <div className={styles.detailsActions}>
              {customerProfile && !order && (
                <Button onClick={handleCreateLease} disabled={isCreateLeaseLoading}>
                  {t('unmarkedWsNotices.view.createInvoice')}
                </Button>
              )}
            </div>
          </Grid>
        </CardBody>
      </Card>

      {order && (
        <InvoiceCard
          applicationStatus={noticeDetails.status}
          buttonsRight={
            canDeleteLease(leaseStatus) && (
              <DeleteButton
                buttonText={t('unmarkedWsNotices.view.deleteInvoice')}
                onConfirm={handleDeleteLease}
                disabled={isDeleteLeaseLoading}
              />
            )
          }
          className={styles.fullWidth}
          customerEmail={customerProfile?.primaryEmail ?? null}
          order={order}
          placeDetails={<p>{t('common.terminology.unmarkedWinterStoragePlace')}</p>}
          placeName={noticeDetails.choice.winterStorageAreaName}
          placeProperties={[]}
          placeType={t('common.terminology.unmarkedWinterStoragePlace').toUpperCase()}
          refetchQueries={refetchQueries}
          title={t('common.terminology.invoice').toUpperCase()}
        />
      )}
    </PageContent>
  );
};

export default UnmarkedWsNoticeView;
