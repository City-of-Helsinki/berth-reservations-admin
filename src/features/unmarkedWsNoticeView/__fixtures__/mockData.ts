import {
  UNMARKED_WINTER_STORAGE_NOTICE,
  UNMARKED_WINTER_STORAGE_NOTICE_boatTypes as BOAT_TYPES,
  UNMARKED_WINTER_STORAGE_NOTICE_winterStorageNotice as WINTER_STORAGE_NOTICE,
  UNMARKED_WINTER_STORAGE_NOTICE_winterStorageNotice_customer as CUSTOMER,
} from '../__generated__/UNMARKED_WINTER_STORAGE_NOTICE';
import {
  ApplicationStatus,
  BerthApplicationLanguage,
  CustomerGroup,
  InvoicingType,
  Language,
  OrderStatus,
} from '../../../@types/__generated__/globalTypes';
import { Order } from '../../invoiceCard/types';

export const winterStorageNotice: WINTER_STORAGE_NOTICE = {
  id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox',
  applicationCode: '',
  firstName: 'test',
  lastName: 'test',
  address: 'test',
  municipality: 'helsinki',
  zipCode: '00100',
  phoneNumber: '555555',
  email: 'test@test.com',
  businessId: '',
  companyName: '',
  language: BerthApplicationLanguage.EN,
  customer: null,
  createdAt: '2019-05-06T10:54:21.746387+00:00',
  boatType: '3',
  boatRegistrationNumber: 'test',
  boatWidth: 1,
  boatLength: 1,
  boatName: 'test',
  boatModel: 'test',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: true,
  acceptLibraryNews: true,
  acceptOtherCultureNews: true,
  status: ApplicationStatus.OFFER_GENERATED,
  winterStorageAreaChoices: [
    {
      winterStorageArea: {
        __typename: 'WinterStorageAreaNode',
        id: 'MOCK-WINTER-STORAGE-AREA-0',
        properties: {
          __typename: 'WinterStorageAreaProperties',
          name: 'Iso-Sarvasto',
        },
      },
      priority: 1,
      winterStorageSectionIds: ['MOCK-SECTION'],
      __typename: 'WinterStorageAreaChoiceType',
    },
    {
      winterStorageArea: {
        __typename: 'WinterStorageAreaNode',
        id: 'MOCK-WINTER-STORAGE-AREA-1',
        properties: {
          __typename: 'WinterStorageAreaProperties',
          name: 'Rajasaari',
        },
      },
      priority: 2,
      winterStorageSectionIds: ['MOCK-SECTION'],
      __typename: 'WinterStorageAreaChoiceType',
    },
  ],
  lease: null,
  __typename: 'WinterStorageApplicationNode',
};

export const mockData: UNMARKED_WINTER_STORAGE_NOTICE & {
  winterStorageNotice: WINTER_STORAGE_NOTICE;
  boatTypes: BOAT_TYPES[];
} = {
  winterStorageNotice: winterStorageNotice,
  boatTypes: [
    {
      id: '1',
      name: 'Jollavene',
      __typename: 'BoatTypeType',
    },
    {
      id: '2',
      name: 'Soutuvene',
      __typename: 'BoatTypeType',
    },
    {
      id: '3',
      name: 'Perämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '4',
      name: 'Sisäperämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '5',
      name: 'Keskimoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '6',
      name: 'Purjevene / moottoripursi',
      __typename: 'BoatTypeType',
    },
    {
      id: '7',
      name: 'Troolari',
      __typename: 'BoatTypeType',
    },
    {
      id: '8',
      name: 'Suuri alus (yli 20t)',
      __typename: 'BoatTypeType',
    },
  ],
};

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: '',
  customerGroup: CustomerGroup.PRIVATE,
  firstName: 'Hessu',
  id: '1',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Hopo',
  organization: null,
  primaryAddress: {
    __typename: 'AddressNode',
    id: 'MOCK-ADDRESS',
    address: 'Paratiisitie 13',
    city: 'Ankkalinna',
    postalCode: '00313',
  },
  primaryEmail: { __typename: 'EmailNode', id: 'MOCK-EMAIL', email: 'hessu@hopo.fi' },
  primaryPhone: { __typename: 'PhoneNode', id: 'MOCK-PHONE', phone: '0' },
};

export const mockOrder: Order = {
  fixedProducts: [],
  id: 'MOCK-ORDER',
  optionalProducts: [],
  orderNumber: '1',
  price: 100,
  status: OrderStatus.PAID,
  totalPrice: 100,
};
