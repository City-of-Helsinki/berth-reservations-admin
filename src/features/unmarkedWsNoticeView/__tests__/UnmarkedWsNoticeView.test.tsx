import React from 'react';
import { mount } from 'enzyme';
import { HashRouter } from 'react-router-dom';
import UnmarkedWsNoticeView, { UnmarkedWsNoticeViewProps } from '../UnmarkedWsNoticeView';
import { mockData, mockCustomer, mockOrder } from '../__fixtures__/mockData';
import { getNoticeDetailsData } from '../utils';
import { getCustomerProfile } from '../../customerView/utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';
import DeleteButton from '../../../common/deleteButton/DeleteButton';
import InvoiceCard from '../../invoiceCard/InvoiceCard';
import { canDeleteLease } from '../../../common/utils/leaseUtils';
import { canDeleteApplication } from '../../../common/utils/applicationUtils';
import StatusLabel from '../../../common/statusLabel/StatusLabel';

const mockNoticeDetails = getNoticeDetailsData(mockData.winterStorageNotice, mockData.boatTypes);

const mockProps: UnmarkedWsNoticeViewProps = {
  customerProfile: null,
  order: null,
  leaseStatus: null,
  noticeDetails: mockNoticeDetails,
  winterStorageNotice: mockData.winterStorageNotice,
  isDeletingNotice: false,
  isCreateLeaseLoading: false,
  isDeleteLeaseLoading: false,
  refetchQueries: [],
  handleUnlinkCustomer: jest.fn(),
  handleCreateLease: jest.fn(),
  handleDeleteNotice: jest.fn(),
  handleDeleteLease: jest.fn(),
  handleEditCustomer: jest.fn(),
  handleLinkCustomer: jest.fn(),
};

// LinkApplicationToCustomer is mocked to limit the test scope
jest.mock('../../linkApplicationToCustomer/LinkApplicationToCustomerContainer', () => {
  const LinkApplicationToCustomerContainer = () => <div>LinkApplicationToCustomerContainer</div>;

  return {
    __esModule: true,
    default: LinkApplicationToCustomerContainer,
  };
});

describe('UnmarkedWsNoticeView', () => {
  const getWrapper = (props?: Partial<UnmarkedWsNoticeViewProps>) =>
    mount(
      <HashRouter>
        <UnmarkedWsNoticeView {...mockProps} {...props} />
      </HashRouter>
    );

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally with customer', () => {
    const customerProfile = getCustomerProfile(mockCustomer);
    const wrapper = getWrapper({ customerProfile });

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders invoice card correctly if a lease has been created', () => {
    const wrapper = getWrapper({ customerProfile: getCustomerProfile(mockCustomer), order: mockOrder });

    expect(wrapper.find(StatusLabel)).toBeDefined();
    expect(wrapper.find(InvoiceCard)).toBeDefined();
  });

  it('renders delete notice button if notice can be deleted', () => {
    Object.values(ApplicationStatus)
      .filter(canDeleteApplication)
      .forEach((status) => {
        const wrapper = getWrapper({ ...mockProps, noticeDetails: { ...mockNoticeDetails, status } });
        expect(wrapper.find('.actionsRight').find(DeleteButton).length).toBe(1);
      });
  });

  it('does not render delete notice button if notice cannot be deleted', () => {
    Object.values(ApplicationStatus)
      .filter((status) => !canDeleteApplication(status))
      .forEach((status) => {
        const wrapper = getWrapper({ ...mockProps, noticeDetails: { ...mockNoticeDetails, status } });
        expect(wrapper.find('.actionsRight').find(DeleteButton).length).toBe(0);
      });
  });

  it('renders delete lease button if lease can be deleted', () => {
    Object.values(LeaseStatus)
      .filter(canDeleteLease)
      .forEach((status) => {
        const wrapper = getWrapper({
          ...mockProps,
          order: mockOrder,
          leaseStatus: status,
        });
        expect(wrapper.find(InvoiceCard).find(DeleteButton).length).toBe(1);
      });
  });

  it('does not render delete lease button if lease cannot be deleted', () => {
    Object.values(LeaseStatus)
      .filter((status) => !canDeleteLease(status))
      .forEach((status) => {
        const wrapper = getWrapper({
          ...mockProps,
          order: mockOrder,
          leaseStatus: status,
        });
        expect(wrapper.find(InvoiceCard).find(DeleteButton).length).toBe(0);
      });
  });
});
