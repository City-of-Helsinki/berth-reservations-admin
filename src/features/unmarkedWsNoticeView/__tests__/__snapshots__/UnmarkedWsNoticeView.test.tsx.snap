// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnmarkedWsNoticeView renders normally 1`] = `
<div
  class="pageContent noticeView"
>
  <div
    class="fullWidth"
  >
    <h2
      class="text standard h2 xl normalWeight pageTitle"
    >
      NOSTOJÄRJESTYSILMOITUS
    </h2>
  </div>
  <div
    class="actions"
  >
    <div
      class="noticeStatus"
    >
      <h2
        class="text standard h2 xl normalWeight"
      >
        Ilmoitus 06.05.2019
      </h2>
      <span
        class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_info__3YrAe status-label_hds-status-label--info__J28-H"
      >
        Käsitelty
      </span>
    </div>
    <div
      class="actionsRight"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Poista ilmoitus
        </span>
      </button>
    </div>
  </div>
  <div>
    LinkApplicationToCustomerContainer
  </div>
  <div
    class="card fullWidth"
  >
    <div
      class="cardHeader"
    >
      <span
        class="title"
      >
        NOSTOJÄRJESTYSILMOITUS
      </span>
    </div>
    <div
      class="cardBody"
    >
      <div
        class="grid cols3"
      >
        <div>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              ILMOITUS
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Alue:
                </span>
                <span
                  class="value left"
                >
                  Iso-Sarvasto
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Vastaanotettu:
                </span>
                <span
                  class="value left"
                >
                  06.05.2019 klo 10.54
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Status:
                </span>
                <span
                  class="value left"
                >
                  Käsitelty
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              OMISTAJAN TIEDOT
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Etunimet:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Sukunimi:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Jakeluosoite:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Postinumero:
                </span>
                <span
                  class="value left"
                >
                  00100
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Postitoimipaikka:
                </span>
                <span
                  class="value left"
                >
                  helsinki
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Matkapuhelin:
                </span>
                <span
                  class="value left"
                >
                  555555
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Sähköpostiosoite:
                </span>
                <span
                  class="value left"
                >
                  test@test.com
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Asiakasryhmä:
                </span>
                <span
                  class="value left"
                >
                  Yksityinen
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Asiointikieli:
                </span>
                <span
                  class="value left"
                >
                  Englanti
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Huomiot:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
            </section>
          </article>
        </div>
        <div>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              VENEEN TIEDOT
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen tyyppi:
                </span>
                <span
                  class="value left"
                >
                  Perämoottorivene
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Rekisterinumero:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen leveys:
                </span>
                <span
                  class="value left"
                >
                  1,0 m
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen pituus:
                </span>
                <span
                  class="value left"
                >
                  1,0 m
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen syväys:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen paino:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Nimi:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Merkki:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              YHTEENVETO
            </h4>
            <section
              class="body"
            >
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan uutiskirjeen
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa liikuntapalveluista
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa kirjastopalveluista
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa muista kulttuuripalveluista
                </span>
              </div>
            </section>
          </article>
        </div>
        <div
          class="detailsActions"
        />
      </div>
    </div>
     
  </div>
</div>
`;

exports[`UnmarkedWsNoticeView renders normally with customer 1`] = `
<div
  class="pageContent noticeView"
>
  <div
    class="fullWidth"
  >
    <h2
      class="text standard h2 xl normalWeight pageTitle"
    >
      NOSTOJÄRJESTYSILMOITUS
    </h2>
  </div>
  <div
    class="actions"
  >
    <div
      class="noticeStatus"
    >
      <h2
        class="text standard h2 xl normalWeight"
      >
        Ilmoitus 06.05.2019
      </h2>
      <span
        class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_info__3YrAe status-label_hds-status-label--info__J28-H"
      >
        Käsitelty
      </span>
    </div>
    <div
      class="actionsRight"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Poista ilmoitus
        </span>
      </button>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="cardHeader"
    >
      <span
        class="title"
      >
        ASIAKASTIEDOT
      </span>
      <div
        class="widgets"
      >
        <button>
          <span
            class="text standard span normalWeight"
          >
            Muokkaa
          </span>
        </button>
      </div>
    </div>
    <div
      class="cardBody"
    >
      <article
        class="section"
      >
        <h4
          class="text standard h4 m title"
        >
          HENKILÖTIEDOT
        </h4>
        <section
          class="body"
        >
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Etunimet:
            </span>
            <span
              class="value left"
            >
              <a
                class="internalLink brand"
                href="#/customers/1"
                title="Hessu"
              >
                Hessu
              </a>
            </span>
          </div>
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Sukunimi:
            </span>
            <span
              class="value left"
            >
              <a
                class="internalLink brand"
                href="#/customers/1"
                title="Hopo"
              >
                Hopo
              </a>
            </span>
          </div>
        </section>
      </article>
      <article
        class="section"
      >
        <section
          class="body"
        >
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Jakeluosoite:
            </span>
            <span
              class="value left"
            >
              Paratiisitie 13
            </span>
          </div>
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Postinumero:
            </span>
            <span
              class="value left"
            >
              00313
            </span>
          </div>
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Postitoimipaikka:
            </span>
            <span
              class="value left"
            >
              Ankkalinna
            </span>
          </div>
        </section>
      </article>
      <article
        class="section"
      >
        <section
          class="body"
        >
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Matkapuhelin:
            </span>
            <span
              class="value left"
            >
              0
            </span>
          </div>
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Sähköpostiosoite:
            </span>
            <span
              class="value left"
            >
              hessu@hopo.fi
            </span>
          </div>
        </section>
      </article>
      <article
        class="section"
      >
        <section
          class="body"
        >
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Asiakasryhmä:
            </span>
            <span
              class="value left"
            >
              Yksityinen
            </span>
          </div>
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Asiointikieli:
            </span>
            <span
              class="value left"
            >
              Suomi
            </span>
          </div>
        </section>
      </article>
      <article
        class="section"
      >
        <section
          class="body"
        >
          <div
            class="labelValuePair"
          >
            <span
              class="label standard"
            >
              Huomiot:
            </span>
            <span
              class="value left"
            >
              -
            </span>
          </div>
        </section>
      </article>
    </div>
     
  </div>
  <div
    class="card actionHistoryCard"
  >
    <div
      class="cardHeader"
    >
      <span
        class="title"
      >
        TOIMINTAHISTORIA
      </span>
    </div>
    <div
      class="cardBody cardBody"
    >
      <article
        class="section section"
      >
        <h4
          class="text standard h4 m title"
        >
          VIIMEAIKAINEN TOIMINTA
        </h4>
        <section
          class="body"
        >
          Placeholder
        </section>
      </article>
      <div
        class="linkRow"
      >
        <a
          class="internalLink brand"
          href="#/"
          title="Näytä koko historia"
        >
          Näytä koko historia
        </a>
      </div>
    </div>
     
  </div>
  <div
    class="card fullWidth"
  >
    <div
      class="cardHeader"
    >
      <span
        class="title"
      >
        NOSTOJÄRJESTYSILMOITUS
      </span>
    </div>
    <div
      class="cardBody"
    >
      <div
        class="grid cols3"
      >
        <div>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              ILMOITUS
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Alue:
                </span>
                <span
                  class="value left"
                >
                  Iso-Sarvasto
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Vastaanotettu:
                </span>
                <span
                  class="value left"
                >
                  06.05.2019 klo 10.54
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Status:
                </span>
                <span
                  class="value left"
                >
                  Käsitelty
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              OMISTAJAN TIEDOT
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Etunimet:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Sukunimi:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Jakeluosoite:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Postinumero:
                </span>
                <span
                  class="value left"
                >
                  00100
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Postitoimipaikka:
                </span>
                <span
                  class="value left"
                >
                  helsinki
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Matkapuhelin:
                </span>
                <span
                  class="value left"
                >
                  555555
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Sähköpostiosoite:
                </span>
                <span
                  class="value left"
                >
                  test@test.com
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Asiakasryhmä:
                </span>
                <span
                  class="value left"
                >
                  Yksityinen
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Asiointikieli:
                </span>
                <span
                  class="value left"
                >
                  Englanti
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Huomiot:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
            </section>
          </article>
        </div>
        <div>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              VENEEN TIEDOT
            </h4>
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen tyyppi:
                </span>
                <span
                  class="value left"
                >
                  Perämoottorivene
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Rekisterinumero:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen leveys:
                </span>
                <span
                  class="value left"
                >
                  1,0 m
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen pituus:
                </span>
                <span
                  class="value left"
                >
                  1,0 m
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen syväys:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Veneen paino:
                </span>
                <span
                  class="value left"
                >
                  -
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Nimi:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
              <div
                class="labelValuePair"
              >
                <span
                  class="label standard"
                >
                  Merkki:
                </span>
                <span
                  class="value left"
                >
                  test
                </span>
              </div>
            </section>
          </article>
          <article
            class="section"
          >
            <h4
              class="text standard h4 m title"
            >
              YHTEENVETO
            </h4>
            <section
              class="body"
            >
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan uutiskirjeen
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa liikuntapalveluista
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa kirjastopalveluista
                </span>
              </div>
              <div>
                <span
                  class="text standard span"
                >
                  Tilaan tietoa muista kulttuuripalveluista
                </span>
              </div>
            </section>
          </article>
        </div>
        <div
          class="detailsActions"
        >
          <button
            class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
            type="button"
          >
            <span
              class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
            >
              Luo lasku
            </span>
          </button>
        </div>
      </div>
    </div>
     
  </div>
</div>
`;
